spring-web/src/main/java/org/springframework/http/converter/HttpMessageConversionException.java

```diff
@@ -17,22 +17,29 @@
 package org.springframework.http.converter;
 
 import org.springframework.core.NestedRuntimeException;
+import org.springframework.http.HttpStatus;
+
+import java.util.Optional;
 
 /**
  * Thrown by {@link HttpMessageConverter} implementations when a conversion attempt fails.
  *
  * @author Arjen Poutsma
+ * @author Sebastien Deleuze
  * @since 3.0
  */
 @SuppressWarnings("serial")
 public class HttpMessageConversionException extends NestedRuntimeException {
 
+	private final Optional<HttpStatus> errorStatus;
```

Comment:
In general, `Optional`s should only be used as a return type, not a field value. So I would store this as a `HttpStatus` field with a potential null value. See http://blog.joda.org/2014/11/optional-in-java-se-8.html for reasons why.